src/Symfony/Component/Notifier/Bridge/LightSms/LightSmsTransport.php

```diff
@@ -0,0 +1,163 @@
+        19 => 'Empty response',
+        20 => 'The number already exists',
+        21 => 'No name',
+        22 => 'Template already exists',
+        23 => 'Missing Month (Format: YYYY-MM)',
+        24 => 'Timestamp not specified',
+        25 => 'Error in access to the list',
+        26 => 'There are no numbers in the list',
+        27 => 'No valid numbers',
+        28 => 'Missing start date (Format: YYYY-MM-DD)',
+        29 => 'Missing end date (Format: YYYY-MM-DD)',
+        30 => 'No date (format: YYYY-MM-DD)',
+        31 => 'Closing direction to the user',
+        32 => 'Not enough money',
+        33 => 'Missing phone number',
+        34 => 'Phone is in stop list',
+        35 => 'Not enough money',
+        36 => 'Can not obtain information about phone',
+        37 => 'Base Id is not set',
+        38 => 'Phone number already exists in this database',
+        39 => 'Phone number does not exist in this database',
+    ];
+
+    public function __construct(string $login, string $password, string $from, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->login = $login;
+        $this->password = $password;
+        $this->from = $from;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('lightsms://%s/external/get/send.php?from=%s', $this->getEndpoint(), $this->from);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $timestamp = time();
+        $data = [
+            'login' => $this->login,
+            'phone' => $this->escapePhoneNumber($message->getPhone()),
+            'text' => $message->getSubject(),
+            'sender' => $this->from,
+            'timestamp' => $timestamp,
+        ];
+        $data['signature'] = $this->generateSignature($data, $timestamp);
+        $endpoint = 'https://'.$this->getEndpoint().'/external/get/send.php?'.http_build_query($data);
+
+        $response = $this->client->request('GET', $endpoint);
+
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
+            throw new TransportException('Unable to send the SMS: ', $response);
+        }
+
+        $content = $response->toArray(false);
+
+        // it happens if the host without www
+        if (isset($content['']) && isset($content['']['error'])) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$content['']['error']], $response);
+        }
+
+        if (isset($content['error'])) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$content['error']], $response);
```

Comment:
It will throw undefined error 🙄 how than validate it 🤔 yes possible they can send another error code or change documentation changes. Maybe put if not exists like as unknown error (add new one number)

Reply:
Opened lightsms specification for sms. Their return all data like strings

```
{"37061122333":{"error":"0", "id_sms":"4092112510348380960001", "cost":"0.5", "count_sms":"1"}}
```
And cost too like string

Reply:
On success their use strings. On failure codes their use integers
```
{"error":6}
```

Reply:
Their uses in different cases different response. How we can manage than 🙄 use (int) before error 🙄

Reply:
Added (int) before return error. On success their return like string. On error their return integer.